staging/src/k8s.io/client-go/tools/leaderelection/metrics.go

```diff
@@ -26,24 +26,26 @@ import (
 type leaderMetricsAdapter interface {
 	leaderOn(name string)
 	leaderOff(name string)
+	slowpathExercised(name string)
 }
 
-// GaugeMetric represents a single numerical value that can arbitrarily go up
-// and down.
-type SwitchMetric interface {
+// LeaderMetric instruments metrics used in leader election.
+type LeaderMetric interface {
```

Comment:
Building a bit on @aojea comment from another PR - changing public interfaces in our official client library is always problematic.
Can we leave the SwitchMetric interface as it was and simply add a new `LeaderMetric` interface as:
```
type LeaderMetric interface {
  SwitchMetric
  SlowPathExercised(name string)
}
```

I'm also thinking about better naming for the method - but don't have a better suggestion.

Reply:
OTOH - we're changing below `MetricsProvider` interface so maybe that's an overkill.
I wouldn't expect people to generally overwrite provider factory and we're fixing all implementations.
So maybe that's actually fine as is.

Reply:
FTR, I keep it as it is.

Reply:
Coming back to this: it is problematic and causes another tight coupling on controller-runtime and client-go which provides its own implementation.

If the original name was kept, even adding the new method would have been more compatible, since it would allow a single consumer to support 1.29 and 1.30. By changing the name, it strictly forces 1.30 usage.

Reply:
This change breaks the upgrade to 1.30 for every controller built on top of controller-runtime e.g. Flux https://github.com/fluxcd/pkg/pull/763. Hopefully the controller-runtime maintainers will do a release soon that deals with this breaking change.

Reply:
Working on https://github.com/kubernetes/kubernetes/pull/124372 to address this. Will cherry pick to 1.30 after merged.